src/super_gradients/common/environment/env_helpers.py

```diff
@@ -142,7 +132,9 @@ def is_main_process():
     """
     if not is_distributed():  # If no DDP, or DDP launching process
         return True
-    elif is_rank_0() and not is_launched_using_sg():  # If DDP launched using torch.distributed.launch or torchrun, we need to run the check on rank 0
+    elif (
+        device_config.requested_rank == 0 and not is_launched_using_sg()
```

Comment:
It's a bit tricky. Basically:
- When launching with SG directly (`python train_from_recipe.py` for instance), we have a launcher job that will call `torch.distributed.launcher.api.elastic_launch` from within `setup_gpu`, which will launch all the DDP subprocesses. This launcher is our "main" process.
- When launching with torchrun/python -m torch.distributed.launch/run, the launcher is not based on SG code so we don't have any control over it (it will not run sanity_check, and we won't be able to save its logs). In that case, our "main" process is the rank0